tests/multi_ptr/multi_ptr_implicit_conversions.h

```diff
@@ -0,0 +1,234 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Provides code for tests for multi_ptr implicit conversions
+//
+*******************************************************************************/
+#ifndef __SYCLCTS_TESTS_MULTI_PTR_IMPLICIT_CONVERSIONS_H
+#define __SYCLCTS_TESTS_MULTI_PTR_IMPLICIT_CONVERSIONS_H
+
+#include "../common/section_name_builder.h"
+#include "multi_ptr_common.h"
+
+namespace multi_ptr_implicit_conversions {
+
+namespace detail {
+
+/**
+ * @brief Functor to invoke implicit conversion
+ * @tparam SrcAccT Source accessor type, enforced to be stated explicitly
+ * @tparam DstAccT Destination accessor type
+ * @param src_acc Instance of a source accessor type
+ */
+template <typename SrcAccT, typename DstAccT>
+struct invoke_implicit_conversion {
+  auto operator()(const SrcAccT &src_acc) const {
+    // We might safely go with returning `src_acc` parameter as `DstAccT` type,
+    // still better to make it explicit
+    const DstAccT result = src_acc;
+    return result;
+  }
+};
+
+/**
+ * @brief Stub functor to use if no implicit conversion is available
+ * @param src_acc Reference to the source accessor instance
+ * @returns The same input reference with no side effects
+ * @details Invocation signature is aligned with `invoke_implicit_conversion`
+ */
+template <typename AccT>
+struct avoid_implicit_conversion {
+  const AccT &operator()(const AccT &src_acc) const {
+    // Just forward parameter further with no copy/move constructor call
+    return src_acc;
+  }
+};
+}  // namespace detail
+
+constexpr int expected_val = 42;
+template <typename T, typename AddrSpaceT, typename IsDecoratedT>
+class run_implicit_convert_tests {
+  using namespace detail;
+  static constexpr sycl::access::address_space address_space =
+      AddrSpaceT::value;
+  static constexpr sycl::access::decorated decorated = IsDecoratedT::value;
+
+  template <typename from_multi_ptr_t, typename dest_multi_ptr_t>
```

Comment:
I think it's better to use either `from`/`to` or `src`/`dest`, but do not mix them.

Reply:
Changed `from` to `src` in template parameters.